### PR TITLE
[ Contact Branch ] Contact Page, Header, Footer, Menu, IE11 Fixes, Minor bug fixes

### PR DESCRIPTION
**Pull request by snacksncode**
Header i Footer branch można chyba usunąć. Any code revisions are welcomed :3

### CODE REVIEW

**Comment by Darex1991** [comment]
@snacksncode @PTISteezy if branches are obsolete please remove them

File: components/Footer/Footer.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        <div className={footerStyles.copyright__content}>
+          <div className={footerStyles.copyright__text}>Copyright © Selleo 2020. Wszelkie prawa zastrzeżone.</div>
+          <SocialIcons/>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+export default Footer;
```
Comment (Darex1991) [inline components/Footer/Footer.js:55]: please set an empty line on the end of file (IDE)

File: components/Footer/Footer.module.scss [hunk 2]
```diff
@@ -0,0 +1,87 @@
+.wrapper {
+    color: #eaeaea;
```
Comment (Darex1991) [inline components/Footer/Footer.module.scss:2]: please use variables

File: components/Footer/Footer.module.scss [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    display: flex;
+    flex-direction: column;
+    background-color: #222222;
```
Comment (Darex1991) [inline components/Footer/Footer.module.scss:5]: 1) variables 
2) please reorder props alphabetically

File: components/Footer/Footer.module.scss [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+.container {
+    max-width: 992px;
+    margin: 0 auto;
+    //display: flex; Mobile IE11 quickfix
```
Comment (Darex1991) [inline components/Footer/Footer.module.scss:11]: you can use `@supports`

Comment (Darex1991) [reply components/Footer/Footer.module.scss:11]: if it is obsolete please remove it

File: components/Footer/Footer.module.scss [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    min-height: 1px;
+    padding: 1rem;
```
Comment (Darex1991) [inline components/Footer/Footer.module.scss:13]: please change it into `px`s

File: components/Footer/Footer.module.scss [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    width: 100%; //quickfix | first column took way more space then it should, overflowing somehow
+    flex-direction: column;
+}
+
+.column {
+    flex-basis: 100%;
+    display: flex;
+    flex-direction: column;
+
```
Comment (Darex1991) [inline components/Footer/Footer.module.scss:22]: ✂️

File: components/Footer/Footer.module.scss [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        img {
+            height: 100%;
+            width: auto;
+        }
+    }
+    .copyright__content {
+        justify-content: space-between !important;
+        flex-direction: row !important;
+    }
+}
```
Comment (Darex1991) [inline components/Footer/Footer.module.scss:87]: in all files:
1) variables
2) px's
3) use BEM/ABEM. So all components will be splitted into separate files. One component. One css files with one parent in css. So there will be ie another component `copyright`
4) empty line on the end of files
5) text className

File: components/FooterBlogEntry/FooterBlogEntry.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import footerBlogStyles from "./FooterBlogEntry.module.scss"
+
+const FooterBlogEntry = ({imageSrc, imageAlt, blogTitle, blogDate}) => {
```
Comment (Darex1991) [inline components/FooterBlogEntry/FooterBlogEntry.js:3]: `const FooterBlogEntry = ({imageSrc, imageAlt, blogTitle, blogDate}) => (`

File: components/FooterBlogEntry/FooterBlogEntry.module.scss [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    &:not(:last-of-type) {
+        margin-bottom: 2rem;
+    }
+    .image {
+        border-radius: 6px;
+        overflow: hidden;
+        margin-bottom: 0.35rem;
+        width: 40%;
+        margin-right: 1rem;
+        img {
```
Comment (Darex1991) [inline components/FooterBlogEntry/FooterBlogEntry.module.scss:14]: class name

File: components/FooterBlogEntry/FooterBlogEntry.module.scss [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    width: 100%;
+    display: flex;
+    min-height: 1px; //IE 11 quickfix (img aspect ratio)
+    &:not(:last-of-type) {
+        margin-bottom: 2rem;
+    }
+    .image {
+        border-radius: 6px;
+        overflow: hidden;
+        margin-bottom: 0.35rem;
```
Comment (Darex1991) [inline components/FooterBlogEntry/FooterBlogEntry.module.scss:11]: without zero before dot

File: components/Header/Header.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react';
+import Link from 'next/link'
```
Comment (Darex1991) [inline components/Header/Header.js:2]: please reorder these

File: components/Header/Header.js [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+import headerStyles from './Header.module.scss'
+import SocialIcons from '../SocialIcons/SocialIcons';
+
+const Header = () => {
+  const [ menuOpened, setMenuOpened ] = useState(false);
+  const toggleMenuState = (state) => {
+    setMenuOpened(oldState => !oldState); //better form of setMenuOpened(!menuOpened)
```
Comment (Darex1991) [inline components/Header/Header.js:9]: what is for this comment?
✂️

File: components/Header/Header.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  return (
+    <>
+      <div className={`
+        ${headerStyles.wrapper}
+        ${menuOpened ? headerStyles.opened : ''}
+      `}>
+        {/* commented as of now. May be revived later
```
Comment (Darex1991) [inline components/Header/Header.js:17]: ✂️

File: components/Header/Header.js [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        ${headerStyles.wrapper}
+        ${menuOpened ? headerStyles.opened : ''}
+      `}>
+        {/* commented as of now. May be revived later 
+        <div className={headerStyles.logo__wrapper}>
+          <Link href="/"><img src='logo.svg'/></Link>
+        </div> */}
+        <div className={headerStyles.navigation}>
+          <div className={headerStyles.menu__title}>Menu</div>
+          <Link href="/"><a className={headerStyles.link}>Start</a></Link>
```
Comment (Darex1991) [inline components/Header/Header.js:23]: please make it more DRY

File: components/Header/Header.js [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+          <Link href="/"><a className={headerStyles.link}>Strefa pacjenta</a></Link>
+          <Link href="/"><a className={headerStyles.link}>Galeria</a></Link>
+          <Link href="/"><a className={headerStyles.link}>Usługi</a></Link>
+          <Link href="/"><a className={headerStyles.link}>O firmie</a></Link>
+          <Link href="/"><a className={headerStyles.link}>Ochrona danych</a></Link>
+          <Link href="/contact"><a className={headerStyles.link}>Kontakt</a></Link>
+          <SocialIcons/>
+        </div>
+      </div>
+      <div onClick={toggleMenuState} className={headerStyles.toggle__button}>
```
Comment (Darex1991) [inline components/Header/Header.js:33]: please move this to another component

File: components/Header/Header.module.scss [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+    background: #fff;
+    display: flex;
+    position: fixed;
+    z-index: 99;
+    top: 0;
+    right: 0;
+    transform: translateX(100%);
+    transition: transform 450ms;
+    transition-timing-function: cubic-bezier(0.83, 0, 0.17, 1);
+    &.opened {
```
Comment (Darex1991) [inline components/Header/Header.module.scss:13]: `&.-opened`

File: components/Header/Header.module.scss [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+        transform: translateX(0%) !important;
```
Comment (Darex1991) [inline components/Header/Header.module.scss:14]: you don't need to have important

File: components/Header/Header.module.scss [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+                    transform: scale(0);
+                }
+                &:nth-child(3) {
+                    transform: rotate(-45deg) translate(5px, -5px);
+                }
+            }
+        }
+    }
+}
+.toggle__button {
```
Comment (Darex1991) [inline components/Header/Header.module.scss:32]: another component

File: components/Header/Header.module.scss [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    position: fixed;
+    top: 1rem;
+    right: 0;
+    transition: background 450ms;
+    transition-timing-function: cubic-bezier(0.83, 0, 0.17, 1);
```
Comment (Darex1991) [inline components/Header/Header.module.scss:41]: I sent you bolilerplate on slack. Please check mixins

Comment (Darex1991) [reply components/Header/Header.module.scss:41]: https://selleo.slack.com/archives/G014YUDB17B/p1591263982003900

File: components/Header/Header.module.scss [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+    width: 40px;
+    height: 40px;
+    background: #fff;
+    box-shadow: 0px 0px 10px rgba(#222, 0.2);
+    z-index: 100;
+    .bar {
```
Comment (Darex1991) [inline components/Header/Header.module.scss:47]: BEM/ABEM

File: components/Header/Header.module.scss [hunk 21]
```diff
@@ -0,0 +1,149 @@
...
+        background: #222;
+        margin: 2px 0;
+        transition: transform 450ms, background 450ms;
+        transition-timing-function: cubic-bezier(0.83, 0, 0.17, 1);
+        &:nth-child(2) {
+            width: 50%;
+        }
+    }
+}
+.logo__wrapper {
```
Comment (Darex1991) [inline components/Header/Header.module.scss:59]: another component

File: components/Header/Header.module.scss [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+    height: 100px;
+    box-shadow: 0 0 15px rgba(#222222, 0.25);
+    z-index: 2;
+    img {
+        width: auto;
+        margin: 0 auto;
+        height: 100%;
+    }
+}
+.navigation {
```
Comment (Darex1991) [inline components/Header/Header.module.scss:71]: again

File: components/Header/Header.module.scss [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+        .link {
+            text-decoration: none;
+            color: #222222;
+            display: flex;
+            padding: .65rem 1rem;
+            align-items: center;
+            justify-content: center;
+            text-align: center;
+        }
+        // .social__links {
```
Comment (Darex1991) [inline components/Header/Header.module.scss:134]: ✂️

File: components/Hero/Hero.js [hunk 24]
```diff
@@ -1,10 +1,19 @@
 import heroStyles from './Hero.module.scss';
 
-const Hero = () => {
+const Hero = ({
+  showAccent, 
+  accentText,
```
Comment (Darex1991) [inline components/Hero/Hero.js:5]: please reorder this

File: components/Layout/Layout.js [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+import Footer from '../Footer/Footer'
+import Hero from '../Hero/Hero'
+
+import layoutStyles from './Layout.module.scss'
+
+const Layout = ({
+    children, 
+    showHero = true, 
+    showAccent = true, 
+    accentText = "Accent Text",
```
Comment (Darex1991) [inline components/Layout/Layout.js:11]: please use defaultProps

File: components/Layout/Layout.js [hunk 26]
```diff
@@ -0,0 +1,38 @@
...
+    showTitle = true,
+    titleText = "Title Text",
+    showSubtitle = true,
+    subtitleText = "Lorem ipsum dolor sit amet consectetur adipisicing elit. Iste soluta repellendus officiis maiores, quidem doloribus labore tempora. Perferendis, nisi neque?"
+}) => {
+    return (
+        <div className={`${layoutStyles.container} ${!showHero ? layoutStyles.heroHidden : ''}`}>
+            <Header/>
+            { 
+                showHero === true &&
```
Comment (Darex1991) [inline components/Layout/Layout.js:21]: `!!showHero`

File: components/Layout/Layout.js [hunk 27]
```diff
@@ -0,0 +1,38 @@
...
+                <Hero 
+                    accentText={accentText} 
+                    titleText={titleText} 
+                    subtitleText={subtitleText}
```
Comment (Darex1991) [inline components/Layout/Layout.js:25]: please reorder these

File: components/Layout/Layout.module.scss [hunk 28]
```diff
@@ -0,0 +1,15 @@
+.container {
+    min-height: 100vh;
+    height: auto;
+}
+.heroHidden {
+    margin-top: 50px;
+}
+.page__content {
```
Comment (Darex1991) [inline components/Layout/Layout.module.scss:8]: BEM

File: components/SocialIcons/SocialIcons.js [hunk 29]
```diff
@@ -0,0 +1,18 @@
+import socialStyles from "./SocialIcons.module.scss"
```
Comment (Darex1991) [inline components/SocialIcons/SocialIcons.js:1]: please reorder these

File: components/SocialIcons/SocialIcons.js [hunk 30]
```diff
@@ -0,0 +1,18 @@
...
+import YoutubeIcon from '../Icons/Youtube'
+import FacebookIcon from '../Icons/Facebook'
+
+const SocialIcons = () => {
```
Comment (Darex1991) [inline components/SocialIcons/SocialIcons.js:5]: const SocialIcons = () => (

File: components/SocialIcons/SocialIcons.js [hunk 31]
```diff
@@ -0,0 +1,18 @@
...
+    return (
+        <div className={socialStyles.wrapper}>
+            <div className={socialStyles.icon}>
+            <a href="https://youtube.com"><YoutubeIcon className={socialStyles.youtube__icon}/></a>
```
Comment (Darex1991) [inline components/SocialIcons/SocialIcons.js:9]: please make it DRY

File: components/SocialIcons/SocialIcons.module.scss [hunk 32]
```diff
@@ -0,0 +1,41 @@
+.wrapper {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    position: relative;
+    margin: auto 1rem 0;
+    // &::before {
```
Comment (Darex1991) [inline components/SocialIcons/SocialIcons.module.scss:7]: ✂️

File: pages/contact.js [hunk 33]
```diff
@@ -0,0 +1,54 @@
...
+const ContactPage = () => {
+  return (
+    <Layout 
+      showHero={true}
+      showAccent={false}
+      titleText="Contact us!"
+      showSubtitle={false}
+    >
+        <h2 className={contactStyles.title}>Bielskie Pogotowie Ratunkowe</h2>
+        <p>
```
Comment (Darex1991) [inline pages/contact.js:14]: please split this component to smaller components

File: styles/contactStyles.module.scss [hunk 34]
```diff
@@ -0,0 +1,17 @@
+.title {
+    font-size: 1.75em;
+    &::after {
```
Comment (Darex1991) [inline styles/contactStyles.module.scss:3]: `&:after {`

File: styles/contactStyles.module.scss [hunk 35]
```diff
@@ -0,0 +1,17 @@
+.title {
+    font-size: 1.75em;
```
Comment (Darex1991) [inline styles/contactStyles.module.scss:2]: `px`

File: styles/contactStyles.module.scss [hunk 36]
```diff
@@ -0,0 +1,17 @@
...
+    &::after {
+        content: "";
+        display: block;
+        width: 3em;
+        height: 0.125em;
+        background: #222;
+        margin-top: .5em;
+    }
+    &.small {
```
Comment (Darex1991) [inline styles/contactStyles.module.scss:11]: .-small

File: styles/globalStyles.scss [hunk 37]
```diff
@@ -0,0 +1,40 @@
+//box sizing
+*, *::after, *::before {
```
Comment (Darex1991) [inline styles/globalStyles.scss:2]: *,
*:after,
*:before

File: styles/globalStyles.scss [hunk 38]
```diff
@@ -0,0 +1,40 @@
...
+}
+//making links look normal
+a {
+    color: currentColor; //follow current color of the text
+    text-decoration: none;
+    &:hover, &:focus {
+        text-decoration: underline;
+    }
+}
+// body { Ah yes IE11 fucked us all up again
```
Comment (Darex1991) [inline styles/globalStyles.scss:25]: ✂️

File: styles/globalStyles.scss [hunk 39]
```diff
@@ -0,0 +1,40 @@
...
+    margin: 0;
+    padding: 0;
+    min-height: 100vh;
+    width: 100%;
+    max-width: 100%;
+    overflow-x: hidden;
+    background-color: #F4F4F4;
+    color: #222;
+}
+//making links look normal
```
Comment (Darex1991) [inline styles/globalStyles.scss:17]: please remove all comments

File: styles/globalStyles.scss [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+//     --background_color_light: #f4f4f4; //main background color
+//     --background_color_dark: #222222; //dark background color
+//     --background_color_darker: #1b1b1b; //darker background color (20%)
+
+//     --text_color_dark: #222222; //main text color
+//     --text_color_light: #eaeaea; //text color on black background
+
+//     --accent: #29b6f6; //accent color
+// }
+::selection {
```
Comment (Darex1991) [inline styles/globalStyles.scss:37]: ✂️  we don't need this

File: components/Footer/Footer.js [hunk 41]
```diff
@@ -0,0 +1,55 @@
+import footerStyles from './Footer.module.scss';
+import SocialIcons from '../SocialIcons/SocialIcons';
+import FooterBlogEntry from '../FooterBlogEntry/FooterBlogEntry';
+
+const Footer = () => {
+  return (
+    <div className={footerStyles.wrapper}>
```
[the remark this answers is not in the document]
Comment (snacksncode) [reply components/Footer/Footer.js:7]: class names are generated e.x if we're talking footer here.
Component doesn't matters in generation. File name matters. Footer.module.scss has class wrapper inside it. So after applying that class we get Footer_wrapper__random. About children i think to do that id just need to make class "some-child" so the result is Footer_some-child. Is that okay?

File: components/Footer/Footer.module.scss (hunk 2 above)
Comment (snacksncode) [reply components/Footer/Footer.module.scss:2]: IE11 ¯\_(ツ)_/¯.
Or do you mean scss $vars

Comment (Darex1991) [reply components/Footer/Footer.module.scss:2]: yes, I wrote that on our slack channel https://selleo.slack.com/archives/G014YUDB17B/p1591877743018100

File: components/Footer/Footer.module.scss (hunk 5 above)
Comment (snacksncode) [reply components/Footer/Footer.module.scss:13]: Sure? 1rem = 16px. That's pretty consistent in all browsers. Will change if needed

File: components/Header/Header.js (hunk 15 above)
Comment (snacksncode) [reply components/Header/Header.js:33]: Sure

File: components/Header/Header.js (hunk 14 above)
Comment (snacksncode) [reply components/Header/Header.js:23]: Care to explain?

File: components/Header/Header.module.scss (hunk 16 above)
Comment (snacksncode) [reply components/Header/Header.module.scss:13]: Class generator... Will look into it.

File: components/Header/Header.module.scss (hunk 17 above)
Comment (snacksncode) [reply components/Header/Header.module.scss:14]: True, will be fixed

File: components/Layout/Layout.js (hunk 25 above)
Comment (snacksncode) [reply components/Layout/Layout.js:11]: Oh, great idea. Will look into this

File: styles/contactStyles.module.scss (hunk 34 above)
Comment (snacksncode) [reply styles/contactStyles.module.scss:3]: yhm CSS2???
------------------ Mozilla MDN
/* CSS3 syntax */
::after

/* CSS2 syntax */
:after
---------------------

File: styles/contactStyles.module.scss (hunk 35 above)
Comment (snacksncode) [reply styles/contactStyles.module.scss:2]: Sure

File: styles/globalStyles.scss (hunk 39 above)
Comment (snacksncode) [reply styles/globalStyles.scss:17]: Should've been gone but whoops 👀

**Comment by snacksncode** [comment]
Branches deleted, CSS will be reviewed. About that whole BEM. Please consider post-generated look of classNames. If that's a problem I can rewrite scss into global file without module generation to meet your needs

File: components/Footer/Footer.js (hunk 41 above)
[the remark this answers is not in the document]
Comment (snacksncode) [reply components/Footer/Footer.js:7]: Okay so. I've looking into modifiers in BEM with [S]CSS Modules. I can make it work. But with
block__element_modifier so i can use `className={${styles.element} ${styles.element_modifier}}` or use the proper block__element--modifier resulting in code: `className={${styles.element} ${styles['element--modifier']}}` is that acceptable?

[the remark this answers is not in the document]
Comment (snacksncode) [reply components/Footer/Footer.js:7]: Which one do you prefer

File: components/Footer/Footer.module.scss (hunk 5 above)
Comment (Darex1991) [reply components/Footer/Footer.module.scss:13]: yes please

File: components/Header/Header.js (hunk 14 above)
Comment (Darex1991) [reply components/Header/Header.js:23]: Firstly you can move all navigation to another component
Next, you can create a function or component with links and iterate on them.

**Pull request merged**